csharp/sources/Valkey.Glide.InterOp/NativeClient.cs

```diff
@@ -0,0 +1,509 @@
+        TaskCompletionSource<Value> tcs = new(TaskCreationOptions.RunContinuationsAsynchronously);
+
+        GCHandle dataHandle = GCHandle.Alloc(tcs, GCHandleType.Normal);
+        var bytes = new List<nint>();
+        try
+        {
+            byte*[] argsArr = new byte*[args.Length];
+            try
+            {
+                if (args.Length <= SmallStringOptimizationArgs)
+                {
+                    for (int i = 0; i < args.Length; i++)
+                    {
+                        // ReSharper disable once StackAllocInsideLoop
+                        // We do this intentionally here in a "low allocation" (max: 20 * 100 bytes) environment
+                        byte* buffer = stackalloc byte[SmallStringOptimizationBuffer];
+                        byte* ptr = MarshalBytes(args[i], buffer, SmallStringOptimizationBuffer);
+                        argsArr[i] = ptr;
+                    }
+                }
+                else
+                {
+                    for (int i = 0; i < args.Length; i++)
+                    {
+                        byte* ptr = MarshalBytes(args[i]);
+                        argsArr[i] = ptr;
+                    }
+                }
+
+                CommandResult result;
+                var routingInfoNative = routingInfo.ToNative(MarshalCollectingString, MarshalCollectingBytes);
+                if (routingInfoNative is { } routingInfoNativeValue)
+                {
+                    fixed (byte** argsArrPtr = argsArr)
+                        result = Imports.command(
+                            // ReSharper disable once InconsistentlySynchronizedField -- The lock is done to prevent double-free in all scenarios
+                            _handle.Value,
+                            CommandCallbackFptr,
+                            GCHandle.ToIntPtr(dataHandle),
+                            requestType,
+                            &routingInfoNativeValue,
+                            argsArrPtr,
+                            argsArr.Length
+                        );
+                }
+                else
+                {
+                    fixed (byte** argsArrPtr = argsArr)
+                        result = Imports.command(
+                            // ReSharper disable once InconsistentlySynchronizedField -- The lock is done to prevent double-free in all scenarios
+                            _handle.Value,
+                            CommandCallbackFptr,
+                            GCHandle.ToIntPtr(dataHandle),
+                            requestType,
+                            null,
+                            argsArrPtr,
+                            argsArr.Length
+                        );
+                }
+
+                if (result.success != 0 /* is true */)
+                    return tcs.Task;
+                else
+                    throw new Exception(HelperMethods.HandleString(result.error_string));
```

Comment:
All errors should come from failure callback

Reply:
One of the main goals of this PR is to consolidate the various callback systems and adopt full support for the Task-based async model. By leveraging GCHandle (for pinning) and TaskCompletionSource, this change reduces the overhead of manual callback management.
The PR eliminates the old callback system in favor of individualized callbacks.